app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/relationships/RelationshipFragment.java

```diff
@@ -202,6 +201,18 @@ public void goToAddRelationship(@NotNull String teiUid, @NotNull String teiTypeT
         this.startActivityForResult(intent, Constants.REQ_ADD_RELATIONSHIP);
     }
 
+    private void goToSearchTEI(@NonNull String teiUid){
+        Intent intent = new Intent(getContext(), SearchTEActivity.class);
+        Bundle extras = new Bundle();
+        extras.putString("TRACKED_ENTITY_UID", presenter.trackedEntityType(teiUid));
+        extras.putString("PROGRAM_UID", null);
+        Map<String, String> searchableAttributesMap = presenter.teiSearchableAttributeValues(teiUid);
+        extras.putStringArrayList("QUERY_DATA_VALUES", new ArrayList<>(searchableAttributesMap.values()));
+        extras.putStringArrayList("QUERY_DATA_ATTR", new ArrayList<>(searchableAttributesMap.keySet()));
+        intent.putExtras(extras);
+        startActivity(intent);
+    }
+
```

Comment:
I think you could use inline fun <reified T> Activity.navigateTo ? From ActivityExtensions?

Reply:
Seems tt can't be used in a java class